controller/konnect/reconciler_generic_test.go

```diff
@@ -36,21 +95,53 @@ func testNewKonnectEntityReconciler[
 ](
 	t *testing.T,
 	ent T,
+	testCases []konnectEntityReconcilerTestCase,
 ) {
 	t.Helper()
 
-	// TODO: use a mock Konnect SDK factory here and use envtest to trigger real reconciliations and Konnect requests
 	sdkFactory := &ops.MockSDKFactory{}
 
 	t.Run(ent.GetTypeName(), func(t *testing.T) {
-		cl := fakectrlruntimeclient.NewFakeClient()
-		mgr, err := ctrl.NewManager(&rest.Config{}, ctrl.Options{
+		s := scheme.Get()
+		require.NoError(t, configurationv1alpha1.AddToScheme(s))
+		require.NoError(t, configurationv1beta1.AddToScheme(s))
+		require.NoError(t, konnectv1alpha1.AddToScheme(s))
+		cfg := envtest.Setup(t, s)
 
-			Scheme: scheme.Get(),
+		mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+			Scheme: s,
+			Metrics: metricsserver.Options{
+				BindAddress: "0",
```

Comment:
I'd suggest to make a comment that this is set explicitly to disable metrics as we don't need them here.